src/components/MultiHopTrade/helpers.ts

```diff
@@ -0,0 +1,229 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { getDefaultSlippagePercentageForSwapper } from 'constants/constants'
+import type { Asset } from 'lib/asset-service'
+import type { BigNumber } from 'lib/bignumber/bignumber'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+import type { ProtocolFee, SwapperName, TradeQuote } from 'lib/swapper/api'
+import {
+  selectCryptoMarketData,
+  selectFeeAssetById,
+  selectMarketDataByFilter,
+  selectUsdRateByAssetId,
+} from 'state/slices/selectors'
+import { store } from 'state/store'
+import { sumProtocolFeesToDenom } from 'state/zustand/swapperStore/utils'
+
+const convertCryptoBaseUnitToUsdPrecision = (
+  asset: Asset,
+  amountCryptoBaseUnit: BigNumber.Value,
+): BigNumber => {
+  const usdRate = selectUsdRateByAssetId(store.getState(), asset.assetId)
+  if (usdRate === undefined) throw Error(`missing usd rate for assetId ${asset.assetId}`)
+  return bnOrZero(fromBaseUnit(amountCryptoBaseUnit, asset.precision)).times(usdRate)
+}
+
+const getReceiveSideAmountsCryptoBaseUnit = ({
```

Comment:
Does "buy side" here effectively refer to the second hop?

If so, I'm wondering if we should include some "hop" vernacular here to make that more clear.
Maybe `getReceiveHopAmountsCryptoBaseUnit`?

Reply:
Good question, it refers to "last hop AND buy asset AND receive account". Currently the buy side network fee is TODO but later we'll need a check to ensure any network fee that's included here is relevant to the receive account

Reply:
I'll add a comment